src/main/java/org/terasology/math/geom/AABB.java

```diff
@@ -0,0 +1,477 @@
+        Vector3f max = new Vector3f(center);
+        max.add(extent);
+        return new AABB(min, max);
+    }
+
+    public static AABB createEmpty() {
+        return new AABB(new Vector3f(), new Vector3f());
+    }
+
+    /**
+     * Creates a new AABB that encapsulates a set of AABBs
+     *
+     * @param others
+     */
+    public static AABB createEncompassing(Iterable<AABB> others) {
+        Vector3f min;
+        Vector3f max;
+        Iterator<AABB> i = others.iterator();
+        if (i.hasNext()) {
+            AABB next = i.next();
+            min = next.getMin();
+            max = next.getMax();
+        } else {
+            return createEmpty();
+        }
+        while (i.hasNext()) {
+            AABB next = i.next();
+            Vector3f otherMin = next.getMin();
+            Vector3f otherMax = next.getMax();
+            min.min(otherMin);
+            max.max(otherMax);
+        }
+        return new AABB(min, max);
+    }
+
+    public static AABB createEncompasing(TFloatList vertices) {
+        int vertexCount = vertices.size() / 3;
+        if (vertexCount == 0) {
+            return AABB.createEmpty();
+        }
+
+        Vector3f min = new Vector3f(vertices.get(0), vertices.get(1), vertices.get(2));
+        Vector3f max = new Vector3f(vertices.get(0), vertices.get(1), vertices.get(2));
+        for (int index = 1; index < vertexCount; ++index) {
+            min.x = Math.min(min.x, vertices.get(3 * index));
+            max.x = Math.max(max.x, vertices.get(3 * index));
+            min.y = Math.min(min.y, vertices.get(3 * index + 1));
+            max.y = Math.max(max.y, vertices.get(3 * index + 1));
+            min.z = Math.min(min.z, vertices.get(3 * index + 2));
+            max.z = Math.max(max.z, vertices.get(3 * index + 2));
+        }
+        return AABB.createMinMax(min, max);
+    }
+
+    /**
+     * @return The distance from the center to the max node
+     */
+    public Vector3f getExtents() {
+        Vector3f dimensions = new Vector3f(max);
+        dimensions.sub(min);
+        dimensions.scale(HALVING_FACTOR);
+        return dimensions;
+    }
+
+    public Vector3f getCenter() {
+        Vector3f dimensions = new Vector3f(max);
+        dimensions.add(min);
+        dimensions.scale(HALVING_FACTOR);
+        return dimensions;
+    }
+
+    public Vector3f getMin() {
+        return new Vector3f(min);
+    }
+
+    public Vector3f getMax() {
+        return new Vector3f(max);
+    }
+
+    /**
+     * Get a new AABB which have a new location base on the offset
+     * @param offset The offset between the current AABB and the new AABB
+     * @return the new AABB
+     */
+    public AABB move(Vector3f offset) {
+        Vector3f newMin = new Vector3f(min);
+        newMin.add(offset);
+        Vector3f newMax = new Vector3f(max);
+        newMax.add(offset);
+        return new AABB(newMin, newMax);
+    }
+
+    public AABB transform(Quat4f rotation, Vector3f offset, float scale) {
```

Comment:
A javadoc should mention that the rotation part will actually be ignored, since it's an AABB